packages/grid/data-grid/src/tests/layout.DataGrid.test.tsx

```diff
@@ -118,6 +118,29 @@ describe('<DataGrid /> - Layout & Warnings', () => {
         );
         expect(getColumnValues()).to.deep.equal(['Jon Snow', 'Cersei Lannister']);
       });
+
+      it('should support columns.valueGetter even if field param supplied is incorrect resulting in no data to be rendered', () => {
+        const columns = [
+          { field: 'id', hide: true },
+          { field: 'firstName', hide: true },
+          { field: 'lastName', hide: true },
+          {
+            field: 'fullName',
+            valueGetter: (params) => params.getValue('age'), // incorrect field parameter supplied to 'getValue'
+          },
+        ];
+
+        const rows = [
+          { id: 1, lastName: 'Snow', firstName: 'Jon', age: 1 },
+          { id: 2, lastName: 'Lannister', firstName: 'Cersei', age: 2 },
+        ];
+        render(
+          <div style={{ width: 300, height: 300 }}>
+            <DataGrid rows={rows} columns={columns} />
+          </div>,
+        );
+        expect(getColumnValues()).to.deep.equal(['1', '2']);
```

Comment:
Hey @oliviertassinari . I am not sure why this was changed in commit https://github.com/mui-org/material-ui-x/pull/755/commits/8f3bbeb6039d8f11c90bb7500d70c72b20b14c60. 
I had the assertion earlier which ensured that the component did not break if invalid `field` is supplied as told in the `it block` description which was meant to be an error prevention. This means it _works gracefully_ but with empty data.

This is the issue: #749 . Reproducible in [Codesandbox](https://codesandbox.io/s/material-ui-issue-forked-z7s0l?file=/src/Demo.js:381-389)

Now it makes it as a functionality test case, which is already there before just above this test case in `layout.DataGrid.test.tsx`. (`should support columns.valueGetter`)

Reply:
The objective is to tests that it not only doesn't break but return the correct value too.

Reply:
But returning the correct value is already in the previous one, above this test case 🤔 .  This makes it a duplicate test I think.
Perhaps maybe then change the `it block` description from:
```diff
- should support columns.valueGetter even if field param supplied is incorrect resulting in no data to be rendered
+ should support columns.valueGetter even if field param supplied is incorrect
```
?

Reply:
Not breaking and returning the correct value are independent events.

Reply:
It will anyway not break if `rows` has a `age` field!! Like in current version.

Reply:
The fix could have been to silently ignore the error, the previous assertion wouldn't have seen the difference.